### DISCUSSION TITLE
Apply pattern to Multiple Notes

### DISCUSSION THREAD

**Discussion by chrisgrieser** [jglev/obsidian-apply-patterns-plugin#10] [Ideas]
Thanks for the plugin!

I think adding the ability to run the patterns on multiple notes would be really useful. I regularly hear on Discord people mentioning VSCode as a method for batch replacing something. It seems a Obsidian-plugin does not exist yet.

**Comment by chrisgrieser** [comment]
oh and just for convenience, there could also be a "apply pattern to whole document" command. Saves the `cmd/ctrl+ a` to select the whole document ;)

**Reply by jglev** [reply]
Thank you for writing, @chrisgrieser!

> oh and just for convenience, there could also be a "apply pattern to whole document" command. Saves the `cmd/ctrl+ a` to select the whole document ;)

As of [today](https://github.com/publicus/obsidian-apply-patterns-plugin/discussions/4), this is implemented in draft form (but not yet published)! : )

**Comment by jglev** [comment]
> Thanks for the plugin!
> 
> I think adding the ability to run the patterns on multiple notes would be really useful. I regularly hear on Discord people mentioning VSCode as a method for batch replacing something. It seems a Obsidian-plugin does not exist yet.

Thank you for writing!

Global find-and-replace is something that I agree is missing in Obsidian at the moment. With that said, I think that implementing it in this plugin is beyond the scope that I'd like for the plugin to keep. I also hope that the Obsidian core developers will eventually support global find-and-replace, making it unnecessary here. The two elements that give me most pause about implementing it within this plugin are (a) how to best handle undoing changes (what happens if a user [who may not be using version control] makes a change across all files and then realizes that there was a mistake?), and (b) how the interface for such a feature would best work (Would it operate on all files? Would the user set file name filters? Would there be a searchable menu from which the user could select specific files to operate on?).

This is all to say, there are enough open questions that occur to me about supporting cross-document changes that I'd like to table it for now, with the idea that as the plugin gets more use, if it's a feature that's the community continues to need and can't address via VSCode, scripts, etc., the conversation will come up again.

Again, thank you!

**Reply by chrisgrieser** [reply]
I think global find & replace wouldn't be as problematic as you think.
> (a) how to best handle undoing changes (what happens if a user [who may not be using version control] makes a change across all files and then realizes that there was a mistake?)

Wouldn't a big, fat warning message & recommendation to make a backup be enough?

> (b) how the interface for such a feature would best work (Would it operate on all files? Would the user set file name filters? Would there be a searchable menu from which the user could select specific files to operate on?).

You could do something like "Run pattern on every note with the following tag"? Or "Run pattern on every note located inside this folder"?

**Comment by FelipeRearden** [comment]
> Thanks for the plugin!
> 
> I think adding the ability to run the patterns on multiple notes would be really useful. I regularly hear on Discord people mentioning VSCode as a method for batch replacing something. It seems a Obsidian-plugin does not exist yet.

@chrisgrieser 
Maybe a workaround to apply find & replace to multiple notes is to use [Obsidian Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri#search-and-replace) in a QuickAdd Macro using scripts to stack multiples URI from different notes. 

I don't know if it works. It is just an idea
